drake/core/test/matrix_compare_test.cc

```diff
@@ -0,0 +1,152 @@
+#include "drake/util/eigen_matrix_compare.h"
+#include "gtest/gtest.h"
+
+using drake::util::MatrixCompareType;
+
+namespace drake {
+namespace test {
```

Comment:
@david-german-tri: I decided to take your suggestion and use an anonymous namespace to further reduce the number of symbols exported by Drake:

```
namespace drake {
namespace util {
namespace {
```